client/me/purchases/controller.jsx

```diff
@@ -156,7 +156,7 @@ export default {
 		);
 	},
 
-	list() {
+	list( context ) {
```

Comment:
This works perfectly well. However we could also take another approach:
- setup notice which would persist for the next page load
- redirect using `page.redirect` to `/purchases` and display persisted notice
This way we move all those optional logic from `PurchasesList` and place it completely outside. Besides url with notice type won't be stored in the browser history, so when users go back, they won't see page with this notice/message again. Let me know what do you think. Of course it's completely optional for this PR.